pensieve/analysis.py

```diff
@@ -0,0 +1,100 @@
+import mozanalysis.metrics.desktop as mmd
+from mozanalysis.experiment import Experiment, TimeLimits
+from mozanalysis.bq import BigQueryContext
+from datetime import datetime, timedelta
+import pytz
+import re
+
+
+class Analysis:
+    """
+    Wrapper for analysing experiments.
+    """
+
+    # todo: where should this come from? experiments, metrics?
+    ANALYSIS_PERIOD = 7  # 1 week
+
+    # list of standard metrics to be computed
+    STANDARD_METRICS = [
+        mmd.active_hours,
+        mmd.uri_count,
+        mmd.ad_clicks,
+        mmd.search_count,
+    ]
+
+    def __init__(self, project, dataset):
+        self.project = project
+        self.dataset = dataset
+        self.bq_context = BigQueryContext(project_id=project, dataset_id=dataset)
+
+    def _current_date(self):
+        """Returns the current date with UTC timezone and time set to 00:00:00."""
+        return datetime.combine(datetime.today(), datetime.min.time()).replace(tzinfo=pytz.utc)
+
+    def _should_analyse_experiment(self, experiment, current_date):
+        """
+        Returns True if a passed experiment should be analysed based
+        on its start date and the last time it got analysed.
+        """
+        date_delta = current_date - experiment.start_date
+        next_analysis_date = current_date + timedelta(days=date_delta.days % self.ANALYSIS_PERIOD)
+
+        return (
+            date_delta.days > 0 and current_date == next_analysis_date
+        ) or experiment.end_date < current_date
```

Comment:
I know you're doing the check in `main()` but let's check `experiment.end_date == current_date` here in case we change that logic.